src/rgw/rgw_lc_s3.cc

```diff
@@ -70,6 +70,9 @@ bool RGWLifecycleConfiguration_S3::xml_end(const char *el) {
     add_rule(rule);
     rule = static_cast<LCRule_S3 *>(iter.get_next());
   }
+  if (cct->_conf->rgw_lc_max_rules < rule_map.size()) {
```

Comment:
It seems like there would need to be some warning about this situation?

Reply:
Done